src/GetSystemMetrics-Hook.cpp

```diff
@@ -19,23 +69,457 @@ int WINAPI Hooked_GetSystemMetrics(int nIndex)
+                VirtualProtect(targetEmptyClipboard, sizeof(originalBytesForEmptyClipboard), oldProtect, &oldProtect);
+            }
+        }
+    }
+
+    // Hook GetForegroundWindow
+    if (hUser32)
+    {
+        void *targetGetForegroundWindow = GetProcAddress(hUser32, "GetForegroundWindow");
+        if (targetGetForegroundWindow)
+        {
+            DWORD jumpGetForeground = (DWORD)MyGetForegroundWindow - (DWORD)targetGetForegroundWindow - 5;
+            memcpy(originalBytesForGetForeground, targetGetForegroundWindow, sizeof(originalBytesForGetForeground));
+            if (VirtualProtect(targetGetForegroundWindow, sizeof(originalBytesForGetForeground), PAGE_EXECUTE_READWRITE, &oldProtect))
+            {
+                *((BYTE *)targetGetForegroundWindow) = 0xE9;
+                *((DWORD *)((BYTE *)targetGetForegroundWindow + 1)) = jumpGetForeground;
+                VirtualProtect(targetGetForegroundWindow, sizeof(originalBytesForGetForeground), oldProtect, &oldProtect);
+            }
+        }
+    }
+
+    // Hook TerminateProcess
+    HMODULE hKernel32 = GetModuleHandleA("kernel32.dll");
+    if (hKernel32)
+    {
+        void *targetTerminateProcess = GetProcAddress(hKernel32, "TerminateProcess");
+        if (targetTerminateProcess)
+        {
+            DWORD jumpTerminateProcess = ((DWORD)MyTerminateProcess - (DWORD)targetTerminateProcess - 5);
+            memcpy(originalBytesForTerminateProcess, targetTerminateProcess, sizeof(originalBytesForTerminateProcess));
+            if (VirtualProtect(targetTerminateProcess, sizeof(originalBytesForTerminateProcess), PAGE_EXECUTE_READWRITE, &oldProtect))
+            {
+                *((BYTE *)targetTerminateProcess) = 0xE9;
+                *((DWORD *)((BYTE *)targetTerminateProcess + 1)) = jumpTerminateProcess;
+                VirtualProtect(targetTerminateProcess, sizeof(originalBytesForTerminateProcess), oldProtect, &oldProtect);
+            }
+        }
+    }
+
+    // Hook ExitProcess
+    if (hKernel32)
+    {
+        void *targetExitProcess = GetProcAddress(hKernel32, "ExitProcess");
+        if (targetExitProcess)
+        {
+            DWORD jumpExitProcess = ((DWORD)MyExitProcess - (DWORD)targetExitProcess - 5);
+            memcpy(originalBytesForExitProcess, targetExitProcess, sizeof(originalBytesForExitProcess));
+            if (VirtualProtect(targetExitProcess, sizeof(originalBytesForExitProcess), PAGE_EXECUTE_READWRITE, &oldProtect))
+            {
+                *((BYTE *)targetExitProcess) = 0xE9;
+                *((DWORD *)((BYTE *)targetExitProcess + 1)) = jumpExitProcess;
+                VirtualProtect(targetExitProcess, sizeof(originalBytesForExitProcess), oldProtect, &oldProtect);
+            }
+        }
+    }
+
+    // create a message box to show the dll is loaded
+    MessageBoxA(NULL, "Injected :)", "UndownUnlock", MB_OK);
+}
 
-BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+void InstallFocus()
```

Comment:
I think that I've pointed out most of the problematic code below, but here's some general reading about what is and isn't safe to do inside of `DllMain`:

- https://www.google.com/search?q=site%3Adevblogs.microsoft.com%2Foldnewthing+dllmain
- https://learn.microsoft.com/en-us/windows/win32/dlls/dllmain
- https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-best-practices#general-best-practices